app/models/user.rb

```diff
@@ -7,6 +7,51 @@ class User < ApplicationRecord
   validates :name, presence: true, length: { maximum: 20 }
 
   has_many :posts
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships
+  has_many :reverse_friendship, class_name: 'Friendship', foreign_key: 'friend_id'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :pending_request, -> { where confirm: false }, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def add_friend(another_user)
+    friendships.create(friend_id: another_user)
+  end
+
+  def pending_request
+    results = []
+    reverse_friendship.each do |request|
+      results << request.user if request.confirm == false
+    end
+    results
+  end
```

Comment:
Even if it has passed, It's not good to use a name twice in a class mostly in such a situation.